### ISSUE TITLE
Moq Mock should use values instead of `It.Is<>(x == X)`

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#942]
Using Moq's `It.Is<>(x == X)` matchers for concrete values should cause a violation.
Instead of comparing it via `It.Is` matchers the values should be directly provided.

Example:
Instead of:
```c#
Mock.Verify(_ => _.DoSomething(It.Is<string>(s => s == "whatever")));
```
we should write:
```c#
Mock.Verify(_ => _.DoSomething("whatever"));
```
This makes the code easier to read.

**Note:** We should _NOT_ report a violation in case there are properties invoked within the matcher, such as in:
```c#
Mock.Verify(_ => _.DoSomething(It.Is<Data>(d => d.Value == "whatever")));
```
